index.js

```diff
@@ -101,20 +123,33 @@ const modifyManifest = (manifest, manifestPath, runTests = false) => {
  * @param {string} runtimeRepo The repository used to install the runtime from
  * @param {string} buildDir Where to build the application
  * @param {string} repoName The flatpak repository name
+ * @param {boolean} cacheBuildDir Whether to enable caching the build directory
+ * @param {string} cacheKey The key used to cache the build directory
  */
-const build = async (manifest, manifestPath, bundle, runtimeRepo, buildDir, repoName) => {
+const build = async (manifest, manifestPath, bundle, runtimeRepo, buildDir, repoName, cacheBuildDir, cacheKey) => {
     const appId = manifest["app-id"] || manifest["id"]
     const branch = manifest["branch"] || core.getInput("branch") || "master"
 
     core.info("Building the flatpak...")
 
-    await exec.exec(`xvfb-run --auto-servernum flatpak-builder`, [
+    let args = [
         `--repo=${repoName}`,
         "--disable-rofiles-fuse",
         "--install-deps-from=flathub",
-        buildDir,
-        manifestPath,
-    ])
+        "--force-clean",
+    ]
+    if (cacheBuildDir) {
+        args.push("--ccache")
+    }
+    args.push(...[buildDir, manifestPath])
```

Comment:
The whole array decomposition seems unnecessary here (i.e. I think it can be written as `args.push(buildDir, manifestPath)`)

Reply:
aha, interesting. I wasn't aware of such thing. Thanks!